vowpalwabbit/parse_args.cc

```diff
@@ -83,13 +83,16 @@ license as described in the file LICENSE.
 #include "options_boost_po.h"
```

Comment:
This could use help.

Reply:
We should not bring string, equal, vector into the global namespace. We don't want a repeat of the minmax issues, namespaces guard us against ambiguity

Reply:
Is this a "real" problem here?  In other words, is string or vector defined anywhere else in the codebase or in the include dependencies?

Reply:
I dont think this is a problem in our code base right now, but common object and function names like string or equal typically get overloaded eventually. Since the version you use is entirely dependent on the linker, that can result in really hard-to-find bugs in unrelated code.

The most common and least ambiguous rule I see is "don't 'using' anything in libraries". But there shouldn't be any issues 'using' unusual symbols like cout or endl